base/siteorigin-widget.class.php

```diff
@@ -727,23 +727,26 @@ private function save_css( $instance ){
 
 		$css = $this->get_instance_css($instance);
 
-		if( !empty($css) ) {
-
+		if ( ! empty( $css ) ) {
 			if ( WP_Filesystem() ) {
 				global $wp_filesystem;
 				$upload_dir = wp_upload_dir();
 
 				if ( ! $wp_filesystem->is_dir( $upload_dir['basedir'] . '/siteorigin-widgets/' ) ) {
-					$wp_filesystem->mkdir( $upload_dir['basedir'] . '/siteorigin-widgets/' );
+					$store_css = $wp_filesystem->mkdir( $upload_dir['basedir'] . '/siteorigin-widgets/' );
 				}
+				
+				if ( empty( $store_css ) ) {
```

Comment:
The logic needs a bit more tweaking. This will skip trying to write the file if the `siteorigin-widgets` directory already existed and it will try to write the file if the attempt to create the directory failed.